vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java

```diff
@@ -680,4 +693,163 @@ public static Stream<Locale> getSupportedAvailableLocales() {
     private static String format(LocalTime time) {
         return time != null ? time.toString() : null;
     }
+
+    /**
+     * Gets the internationalization object previously set for this component.
+     * <p>
+     * NOTE: Updating the instance that is returned from this method will not
+     * update the component if not set again using
+     * {@link #setI18n(TimePickerI18n)}
+     *
+     * @return the i18n object or {@code null} if no i18n object has been set
+     */
+    public TimePickerI18n getI18n() {
+        return i18n;
+    }
+
+    /**
+     * Sets the internationalization object for this component.
+     *
+     * @param i18n
+     *            the i18n object, not {@code null}
+     */
+    public void setI18n(TimePickerI18n i18n) {
+        this.i18n = Objects.requireNonNull(i18n,
+                "The i18n properties object should not be null");
+    }
+
+    private String getBadInputErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(TimePickerI18n::getBadInputErrorMessage).orElse("");
+    }
+
+    private String getRequiredErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(TimePickerI18n::getRequiredErrorMessage).orElse("");
+    }
+
+    private String getMinErrorMessage() {
+        return Optional.ofNullable(i18n).map(TimePickerI18n::getMinErrorMessage)
+                .orElse("");
+    }
+
+    private String getMaxErrorMessage() {
+        return Optional.ofNullable(i18n).map(TimePickerI18n::getMaxErrorMessage)
+                .orElse("");
+    }
```

Comment:
Nit: as this pattern seems to be something used multiple times not only in this file, but in other components too, would it make sense to extract it as helper function like:

```java
private <T> String getMessageOrEmptyString(T object, Function<T, String> getter) {
    return Optional.ofNullable(object).map(getter).orElse("");
}
```

It would be called as `getMessageOrEmptyString(i18n, TimePickerI18n::getMinErrorMessage)`, for instance.

Reply:
It's a good point. How about this?

```java
private String getI18nErrorMessage(
        Function<DateTimePickerI18n, String> getter) {
    return Optional.ofNullable(i18n).map(getter).orElse("");
}
```

Reply:
For using it just inside of this class, then the generic is not needed 👍🏻

Reply:
Updated.